pkg/apis/edgeapi.cattle.io/v1alpha1/device_template_types.go

```diff
@@ -17,12 +16,15 @@ type DeviceTemplate struct {
 }
 
 type DeviceTemplateSpec struct {
-	DeviceKind     string                `json:"deviceKind,omitempty"`
-	DeviceVersion  string                `json:"deviceVersion,omitempty"`
-	DeviceGroup    string                `json:"deviceGroup,omitempty"`
-	DeviceResource string                `json:"deviceResource,omitempty"`
-	Labels         map[string]string     `json:"labels,omitempty"`
-	TemplateSpec   *runtime.RawExtension `json:"templateSpec,omitempty"`
+	DeviceKind     string            `json:"deviceKind,omitempty"`
+	DeviceVersion  string            `json:"deviceVersion,omitempty"`
+	DeviceGroup    string            `json:"deviceGroup,omitempty"`
+	DeviceResource string            `json:"deviceResource,omitempty"`
+	Labels         map[string]string `json:"labels,omitempty"`
+
+	DisplayName         string `json:"displayName"`
```

Comment:
> why not use the metadata name as its display name?

Since the metadata name has a fixed format restriction, it may not be flexible enough to meet the user's naming conventions.